### PR TITLE
add public routes, public user search

### PR DESCRIPTION
**Pull request by AlexAzalor**

### CODE REVIEW

File: api/controllers/user.py [hunk 1]
```diff
@@ -151,3 +151,78 @@ def get_user_profile(user_uuid: str, lang: Literal[CFG.UA, CFG.EN], db: Session)
         locations=locations,
         owned_rates_count=db_user.owned_rates_count,
     )
+
+
+def public_search_users(query: s.UserSearchIn, db: Session) -> s.PublicUsersSearchOut:
```
Comment (Taraslysak) [inline api/controllers/user.py:157]: What is the difference between public and non-public search?

File: app/models/user.py [hunk 2]
```diff
@@ -29,6 +29,12 @@ class User(db.Model, ModelMixin):
         sa.CheckConstraint(f"average_rate <= {CFG.MAXIMUM_RATE}", name="max_rate_check"),
     )
 
+    # avatar
+    # gender? for avatar
+    # role?
+    # description
+    # Досвід&Сертифікати
```
Comment (Taraslysak) [inline app/models/user.py]: ?????

File: api/routes/user.py [hunk 3]
```diff
@@ -51,3 +51,30 @@ def get_user_profile(
...
+    """Returns filtered list of users"""
+
+    result = c.public_search_users(query, db)
+    return result
+
+
+@user_router.get("/public/{user_uuid}", status_code=status.HTTP_200_OK, response_model=s.PublicUserProfileOut)
+def get_public_user_profile(
+    user_uuid: str,
+    lang: Literal[CFG.UA, CFG.EN] = CFG.UA,  # type: ignore
```
Comment (Taraslysak) [inline api/routes/user.py]: Why type ignore?

**Review by Taraslysak — COMMENTED** [review]
Please take a look at my comments

File: api/controllers/user.py (hunk 1 above)
Comment (AlexAzalor) [reply api/controllers/user.py:157]: The difference is in the scheme, in the complexity of the filter and in the public filter we don't work with the current user - "me: m.User"
**def search_users(query: s.UserSearchIn, me: m.User, db: Session)**

File: api/routes/user.py (hunk 3 above)
Comment (AlexAzalor) [reply api/routes/user.py]: I just copy-pasted it. The error says:

Variable not allowed in type expression PylancereportInvalidTypeForm
Type arguments for "Literal" must be None, a literal value (int, bool, str, or bytes), or an enum value PylancereportInvalidTypeForm

Comment (Taraslysak) [reply api/routes/user.py]: This error should no be ignored. May it is a good idea to change it to enum

**Pull request merged**